internal/wire/wire.go

```diff
@@ -89,3 +108,227 @@ func Execute(context interface{}, mgoDB *db.DB, graphDB *cayley.Handle, viewPara
 	log.Dev(context, "Execute", "Completed")
 	return &result, nil
 }
+
+//==============================================================================
+
+// validateStartType verifies the start type of a view path.
+func validateStartType(context interface{}, db *db.DB, v *view.View) error {
+
+	// Extract the first level relationship predicate.
+	var firstRel string
+	var firstDir string
+	for _, segment := range v.Path {
+		if segment.Level == 1 {
+			firstRel = segment.Predicate
+			firstDir = segment.Direction
+		}
+	}
+
+	// Get the relationship metadata.
+	rel, err := relationship.GetByPredicate(context, db, firstRel)
+	if err != nil {
+		return err
+	}
+
+	// Get the relevant item types based on the direction of the
+	// first relationship in the path.
+	var itemTypes []string
+	switch firstDir {
+	case "out":
+		itemTypes = rel.SubjectTypes
+	case "in":
+		itemTypes = rel.ObjectTypes
+	}
+
+	// Validate the starting type provided in the view.
+	for _, itemType := range itemTypes {
+		if itemType == v.StartType {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("Start type %s does not match relationship subject types %v", v.StartType, itemTypes)
+}
+
+// viewPathToGraphPath translates the path in a view into a "path"
+// utilized in graph queries.
+func viewPathToGraphPath(v *view.View, key string, graphDB *cayley.Handle) (*path.Path, error) {
+
+	// Sort the view Path value.
+	sort.Sort(v.Path)
+
+	// Loop over the path segments translating the path.
+	var graphPath *path.Path
+	level := 1
+	for _, segment := range v.Path {
+
+		// Check that the level is the level we expect (i.e., that the levels
+		// are in order)
+		if level != segment.Level {
+			err := fmt.Errorf("Invalid view path level, expected %d but seeing %d", level, segment.Level)
+			return graphPath, err
+		}
+
+		// Initialize the path, if we are on level 1.
+		if level == 1 {
+
+			// Add the first level relationship.
+			switch segment.Direction {
+			case "in":
```

Comment:
Would it not make sense to move the strings `in` and `out` into package const's?